### ISSUE TITLE
Slow refresh times while loading structures

### ISSUE THREAD

**Issue by bwhittington** [resource/fabricator#103]
I've been working through some structures that contain their own HTML, Base64 encoded images, and a few Markdown references to components and I am experiencing some very long refresh times when saving my changes.  If I review what node is telling me, it stops at the collate step and then after a minute it continues down its refresh path.  Another entry says 'collate' after 1.67 min.  I was wondering if there was anything I might be doing wrong?  I would be willing to send my fabricator project along if that is what it takes. 

Other than it being really slow, I haven't really run into other problems.

**Comment by LukeAskew** [comment]
Could be the base64 -- I've not tested it with a large file.

Could you post the content of your structure to a gist or something?

**Comment by bwhittington** [comment]
It definitely hates base64 encoded images when refreshing structures and templates.  I began removing them one by one from my code and each time the time went down pretty significantly.  I was using close to 10 images with 500KB total which doesn't seem like too much to me.  A 4 KB one was adding 4 seconds to the 'collate' task.  After removing all of these encoded images I went from 1.67 min to 10s.

Not sure if anything can be done about it but it seems like base64 encoding things is not the way to go in fabricator.
